File: CBToast_OC/CBToast/BOViewController.m
//
//  BOViewController.m
//  CBToast
//
//  Created by chang on 2018/3/29.
//  Copyright © 2018年 chang. All rights reserved.
//

#import "BOViewController.h"
#import "CBToast.h"

@interface BOViewController ()

@end

@implementation BOViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.view.backgroundColor = [UIColor whiteColor];
    
    CGFloat main_width = [UIScreen mainScreen].bounds.size.width;
    CGFloat main_height = [UIScreen mainScreen].bounds.size.height;
    
    UIButton *btn = [UIButton buttonWithType:UIButtonTypeCustom];
    btn.frame = CGRectMake(30, 90, main_width-30*2, 40);
    btn.tag = 1;
    btn.selected = YES;
    btn.backgroundColor = [UIColor orangeColor];
    btn.titleLabel.font = [UIFont systemFontOfSize:14];
    [btn setTitle:@"正在加载中" forState:UIControlStateNormal];
    [btn addTarget:self action:@selector(showToastAction:) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:btn];
    
    
    
    UIButton *bottomBtn = [UIButton buttonWithType:UIButtonTypeCustom];
    bottomBtn.frame = CGRectMake(30, main_height-40-30, main_width-30*2, 40);
    bottomBtn.tag = 100;
    bottomBtn.backgroundColor = [UIColor orangeColor];
    bottomBtn.titleLabel.font = [UIFont systemFontOfSize:15];
    [bottomBtn setTitle:@"返回上一页" forState:UIControlStateNormal];
    [bottomBtn addTarget:self action:@selector(showToastAction:) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:bottomBtn];
}

- (void)showToastAction:(UIButton *)sender
{
    switch (sender.tag) {
        case 1:
        {
            if (sender.selected) {
                [CBToast showIndicatorToastAction:sender.currentTitle];
            }else{
                [CBToast hiddenIndicatorToastAction];
            }
            sender.selected = !sender.selected;
        }
            break;
            
        case 100:
        {
            [self dismissViewControllerAnimated:YES completion:nil];
        }
            break;
        default:
            break;
    }
}



- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: CBToast_Swift/CBToast_Swift/BOViewController.swift
//
//  Created by c on 2020/7/25.
//  Copyright © 2020 c. All rights reserved.
//

import UIKit

class BOViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        self.view.backgroundColor = UIColor.white

        let btn = UIButton.init(type: .custom)
        btn.frame = CGRect.init(x: 30, y: 90, width: main_width-30*2, height: 40)
        btn.tag = 1
        btn.isSelected = true
        btn.backgroundColor = UIColor.orange
        btn.titleLabel?.font = UIFont.systemFont(ofSize: 14)
        btn.setTitle("正在加载中", for: .normal)
        btn.addTarget(self, action: #selector(showToastAction(sender:)), for: .touchUpInside)
        self.view.addSubview(btn)
     
        let bottomBtn = UIButton.init(type: .custom)
        bottomBtn.frame = CGRect.init(x: 30, y: main_height-40-30, width: main_width-30*2, height: 40)
        bottomBtn.tag = 100
        
        bottomBtn.backgroundColor = UIColor.orange
        bottomBtn.titleLabel?.font = UIFont.systemFont(ofSize: 15)
        bottomBtn.setTitle("返回上一页", for: .normal)
        bottomBtn.addTarget(self, action: #selector(showToastAction(sender:)), for: .touchUpInside)
        self.view.addSubview(bottomBtn)
    }

    @objc func showToastAction(sender : UIButton) {
        switch sender.tag {
        case 1:
            if sender.isSelected == true {
                CBToast.showIndicatorToastAction(message: sender.currentTitle! as NSString)
            }else{
                CBToast.hiddenIndicatorToastAction()
            }
            sender.isSelected = !sender.isSelected
        case 100:
            self.dismiss(animated: true, completion: nil)
        default:
            break
        }
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

}
